pkg/plugins/golang/multi-module/v1alpha1/api.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+
+	"sigs.k8s.io/kubebuilder/v3/pkg/config"
+	"sigs.k8s.io/kubebuilder/v3/pkg/machinery"
+	"sigs.k8s.io/kubebuilder/v3/pkg/model/resource"
+	"sigs.k8s.io/kubebuilder/v3/pkg/plugin"
+	"sigs.k8s.io/kubebuilder/v3/pkg/plugin/util"
+	"sigs.k8s.io/kubebuilder/v3/pkg/plugins/golang/multi-module/v1alpha1/scaffolds"
+)
+
+var _ plugin.CreateAPISubcommand = &createAPISubcommand{}
+
+type createAPISubcommand struct {
+	config config.Config
+
+	resource *resource.Resource
+}
+
+func (p *createAPISubcommand) UpdateMetadata(cliMeta plugin.CLIMetadata, subcmdMeta *plugin.SubcommandMetadata) {
+	subcmdMeta.Description = `Scaffold a Kubernetes API by writing a Resource definition and/or a Controller.
+
+If information about whether the resource and controller should be scaffolded
+was not explicitly provided, it will prompt the user if they should be.
+
+After the scaffold is written, the dependencies will be updated and
+make generate will be run.
+
+Warning: This will also create multiple go.mod files. If you are not careful, you can break your dependency chain.
+The multi-module extension will create replace directives for local development, 
+which you might want to drop after creating your first stable API.
```

Comment:
@jakobmoellersap you can create only once right?
We can check and only create if not exist

Reply:
Is that not desired? I would imagine if you create APIs you do not want to switch between multi-module or not. My idea would have been that it would only need to be setup once in the init. The go.mod is using create if not exist with `f.IfExistsAction = machinery.SkipFile`

Reply:
So, are you saying that is possible to have a project that is multi-module and not export all apis? Should not all apis be replaced in this case?

Reply:
Ehm actually I tried to say the opposite :D sorry for my bad wording here. This is also why im gonna remove the resource from the project config, I think it should be an `all-or-nothing` for the APIs

Reply:
Hi @jakobmoellersap, 

Cool. So, if it is all all-or-nothing then we should not need to explicitly call the plugin every time that we run create api. We should only:

a) Call the plugin for init or edit so we enable that
b) Therefore, after that when we call create api it will be tracked in the plugin layout (such as we have the declarative one) which means that when we run create api the CLI will automatically  call all create api subcommand implemented for all plugin in the layout in a chain

Also, that means we do not need to track the API resources for this plugin since it has no specific behaviour for each API. That is valid for the whole project after you use that.

By last in case we use the edit command, we might want to try to change the default scaffold with what was done before to fix all. However, if an error is faced would be great if we raise that in the logs and let users know what they need to do manually. 

Is that make sense?

Reply:
For case B i spotted a difference in setting the layout with `init` vs `create api`. In create API, the plugin is not tracked in "layout" but only in "plugins". This means that you would have to continously call the plugin for every create call.

Reply:
But why we would call the plugin (informing it the --plugins create API)
Why we would need to track the resources inside of the plugin in this case ( that is only valid for the cases like deploy-image or declarative one where we can have 1 API using the plugin and another not )

Reply:
I think we sorted out closing this one to make eaiser the review